service/s3/s3manager/upload.go

```diff
@@ -373,6 +373,40 @@ func (u Uploader) UploadWithContext(ctx aws.Context, input *UploadInput, opts ..
 	return i.upload()
 }
 
+// UploadWithIterator will upload a batched amount of objects to S3. This operation uses
+// the iterator pattern to know which object to upload next. Since this is an interface this
+// allows for custom defined functionality.
+//
+// Example:
+//	svc:= s3manager.NewUploader(sess)
+//
+//	objects := []BatchUploadObject{
+//		{
+//			Object:	&s3manager.UploadInput {
+//				Key: aws.String("key"),
+//				Bucket: aws.String("bucket"),
+//			},
+//		},
+//	}
+//
+//	iter := &s3managee.UploadObjectsIterator{Objects: objects}
+//	if err := svc.UploadWithIterator(aws.BackgroundContext(), iter); err != nil {
+//		return err
+//	}
+func (u Uploader) UploadWithIterator(ctx aws.Context, iter BatchUploadIterator, opts ...func(*Uploader)) error {
+	var errs []error
+	for iter.Next() {
+		if _, err := u.UploadWithContext(ctx, iter.UploadObject().Object, opts...); err != nil {
+			errs = append(errs, err)
```

Comment:
If returning a list of errors of files that failed to upload, probably want to create a custom error type that wraps the individual error with a type which includes a field for the object key of the upload that failed.